r/README.md

```diff
@@ -0,0 +1,70 @@
+of these features in VSCode as well.
+
+> [!NOTE]
+> A note on the use of the pipe (`|>` or `%>%`) operator, although
+>[very useful for data analysis code](https://r4ds.hadley.nz/workflow-style.html#sec-pipes), 
+>we recommend to avoid their usage in **R package code**. The main reason is that pipes make code 
+>more difficult to debug, especially for very long sequences of pipes, as the intermediate objects
+>are not immediately accessible. In addition, the use of `%>%` would incur an additional dependency.
+
+## Package development
+
+The [*R Packages*](https://r-pkgs.org/) book provides a comprehensive guide to R package development,
+which closely follows the *Tidyverse style guide* (as they're from the same authors). We recommend
+following the best practices described in this book.
+
+To help with adhering to these practices and make your life as an R developer much easier, there are
+the [*devtools*](https://devtools.r-lib.org/) and [*usethis*](https://usethis.r-lib.org/) packages.
+From setting up your package file structure with `usethis::create_package()`, running unit tests
+and other checks with `devtools::check()`, to publishing your package to CRAN with `devtools::release()`,
+these two packages have it all covered. In fact, whenever you need to do anything other than edit
+an `*.R` file, there is probably a [*usethis* function](https://usethis.r-lib.org/reference/index.html)
+function you can (and should!) use instead. And if there is not, ask yourself whether you really need
+to do that thing.
+
+And of course, both these packages play nicely together with RStudio, with several built-in features
+and shortcuts (see the *R Packages* book for tips and tricks).
+
+## CI with GitHub Actions
+
+The [r-lib/actions](https://github.com/r-lib/actions) repository contains several reusable *GitHub Actions* (GHA)
+workfklows for R projects (not only packages). The [examples](https://github.com/r-lib/actions/tree/v2/examples)
+directory provides complete workflows for common R project CI tasks. All of these can be easily set up
+using [`usethis::use_github_action()`](https://usethis.r-lib.org/reference/use_github_action.html).
+
+## Bioconductor
+
+When developing an R package for the [Bioconductor project](https://bioconductor.org/), a few exceptions
+to the above recommendations should be considered. Bioconductor provides
+[their own list of recommendations](https://contributions.bioconductor.org/r-code.html). These are
+largely similar to the *Tidyverse style guide*, with a few exceptions:
+
+* Use of 4 spaces for indenting instead of 2
+* Use of `camelCase` for function names
+
+The reason for the latter is that Bioconductor packages make extensive use of the [S4](https://adv-r.hadley.nz/s4)
+OOP framework in R. Within this framework, the convention is to use `CamelCase` for Class names and
+`camelCase` for method names.
+
+The most important of these rules are checked by
```

Comment:
is there something that autofixes things for you like the 4 spaces?

Reply:
"Fixes" as in replace the 4 spaces with 2 spaces? `styler` can do that, which I just run regularly when working in RStudio. In VSCode, if you have the `vscode-R` plugin installed, the formatting option will also run `styler` and do it automatically by enabling `format on save`

Reply:
I guess what I'm looking for is, is there a maintained config for these tools that configures the rules to what bioc wants?

Reply:
Aah yes, *biocthis* has [`biocthis::bioc_style()`](https://lcolladotor.github.io/biocthis/reference/bioc_style.html) 
I used to have `options(styler.addins_style_transformer = "biocthis::bioc_style()")` in my `~/.Rprofile`, but you could also do this an a per-project basis.

Reply:
🎉 _could_ be worth adding this in to the initial structure

Reply:
Good point, wasn't sure how widely *Bioc* is used in ARC, but I'll add it